cmd/doc.go

```diff
@@ -0,0 +1,37 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+// Commands and Sub-commands
+// =========================
+// Commands need to provide an `Info` method that returns an Info struct.
+
+// The info struct contains: Name, Args, Purpose and a detailed description
+// (Doc).
+// This information is used to provide the default help for the command.
+
+// In the same package, there is `CommandBase` whose purpose is to be composed
+// into new commands, and provides a default no-op SetFlags implementation, a
+// default Init method that checks for no extra args, and a default Help method.
+
+// Supercommands
+// =============
+// `Supercommand`s are commands that do many things, and have "sub-commands" that
+// provide this functionality.  Git and Bazaar are common examples of
+// "supercommands".  Subcommands must also provide the `Command` interface, and
+// are registered using the `Register` method.  The name and aliases are
+// registered with the supercommand.  If there is a duplicate name registered,
+// the whole thing panics.
+
+// Supercommands need to be created with the `NewSuperCommand` function in order
+// to provide a fully constructed object.
+
```

Comment:
suggestion: Add an example of a supercommand. Also mention that they are rare in `juju` -- we mostly don't use that pattern.

Reply:
fixed